### PR TITLE
Contribute Cisco Jabber XMPP connector to botkit

### PR DESCRIPTION
**Pull request by qiongfangzhang**
We open this pull request to contribute Cisco Jabber XMPP connector into botkit framework. People can create Cisco Jabber Bot based on botkit framework, connect to Cisco Webex messenger servers and Cisco IM&Presence servers.  More details for the Jabber Bot can be found in the readme-jabberbot.md and in the attached documents.
[ChatBot-CallFlow-Example.docx](https://github.com/howdyai/botkit/files/1541508/ChatBot-CallFlow-Example.docx)

### CODE REVIEW

File: docs/readme-ciscojabber.md [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+The full code for a simple Cisco Jabber bot is as below and you can find it in ../examples/jabber_bot.js:
+
+~~~ javascript
+    const Botkit = require('./lib/JabberBot.js');
+    var controller = Botkit({
+    json_file_store: './bot_store/'
+    });
+    var bot = controller.spawn({
+    client: {
+        jid: ‘xx@domain.com',
```
Comment (Naktibalda) [inline docs/readme-ciscojabber.md]: Wrong opening quote here.

File: docs/readme-ciscojabber.md [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+~~~
+    
+## Bot Options ##
+When spawn bot from the Botkit controller, there are several options available.
+
+| Argument | Description
+|--- |---
+| jid | Jid of the jabber bot 
+| password | Password of the jabber bot
+| host | host of the Cisco Unified IM & Presence server, no need for bot of Cisco WebEx Messenger
```
Comment (Naktibalda) [inline docs/readme-ciscojabber.md]: If all descriptions start with capital letter, this one should start with `H`.
`not necessary` would look better than `no need`.

File: docs/readme-ciscojabber.md [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+~~~ javascript
+    function ExtractMentionJids(message) {
+    let direct_mention_reg = /href="xmpp:\s?(\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+)\s?"/ig;
+    let email_reg = /\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+/i;
+    let match = message.stanza.toString().match(direct_mention_reg);
+    let mention_jids = [];
+    if (match) {
+    for (let i = 0; i < match.length; i++) {
+    let jid_match = match[i].match(email_reg);
```
Comment (Naktibalda) [inline docs/readme-ciscojabber.md]: Please indent all statements in function body properly.

File: lib/JabberBot.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+            console.log('connection has been closed!');
+        });
+
+        xmpp.on('error', function (err) {
+            console.log(err);
+        });
+
+        xmpp.on('subscribe', function (from) {
+            xmpp.acceptSubscription(from);
+            console.log('accept subscribe from' + from);
```
Comment (Naktibalda) [inline lib/JabberBot.js]: Please add a space between `from` and closing quote `'`.

File: lib/JabberBot.js [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+        {
+            let mention_jids = ExtractMentionJids(message);
+            if (mention_jids.find(findBotJid))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        function ExtractMentionJids(message) {
```
Comment (Naktibalda) [inline lib/JabberBot.js]: `ExtractMentionJids` function is not a constructor so it should start with lowercase `e`.

File: package.json [hunk 6]
```diff
@@ -26,7 +26,10 @@
     "twilio": "^2.11.1",
     "vorpal": "^1.12.0",
     "ware": "^1.3.0",
-    "ws": "^2.2.2"
+    "ws": "^2.2.2",
+	"node-ews": "^3.2.2",
+    "node-xmpp-caps": "0.0.2",
+    "simple-xmpp": "^1.3.0"
```
Comment (Naktibalda) [inline package.json]: Please put new dependencies to alphabetically correct positions.
Putting then at the bottom also creates an unnecessary conflict with #1154.

**Comment by peterswimm** [comment]
This will be included in the next updated of Botkit

**Comment by qiongfangzhang** [comment]
Great to know, thanks Peter for the help on it.

Thanks,
Qiongfang

在 2018年2月10日，上午4:33，Peter Swimm <notifications@github.com<mailto:notifications@github.com>> 写道：


This will be included in the next updated of Botkit

**Pull request merged**